src/jwt/mod.rs

```diff
@@ -20,11 +20,113 @@ new_type![
     JsonWebTokenContentType(String)
 ];
 
+/// Error type used when normalizing [`JsonWebTokenType`] objects
+#[derive(Error, Debug)]
+#[non_exhaustive]
+#[allow(clippy::enum_variant_names)] // new error types might not start with InvalidMime
```

Comment:
now that this is up to 3 variants that are very similar, I think it may be preferable to have a single error type that doesn't distinguish between the different reasons for the MIME type to be invalid. e.g.,
```rust
#[derive(Error, Debug)]
#[error("Invalid JWT type: {typ}")]
pub struct InvalidJsonWebTokenTypeError {
  typ: String,
}
```
this also includes the offending type, which should help users debug the issue. I'm assuming it'll be pretty obvious to users why the printed MIME type isn't valid 🤷‍♂️

Reply:
I thought it might be nice to allow error differentiation here but I see that it is much more simple without it.